javascript-carousel/main.js

```diff
@@ -0,0 +1,65 @@
+var buttonContainer = document.querySelectorAll('.button-container');
+var dataViews = document.querySelectorAll('.view');
+
+function switchViews(string) {
+  for (var i = 0; i < dataViews.length; i++) {
+    if (string === dataViews[i].getAttribute('data-view')) {
+      dataViews[i].className = 'image-container display-flex flex-column justify-center view';
+    } else {
+      dataViews[i].className = 'hidden image-container display-flex flex-column justify-center view';
+    }
+  }
+}
+
+var leftChevron = document.querySelector('.fa-chevron-left');
+var rightChevron = document.querySelector('.fa-chevron-right');
+var currentIndex = 0;
+
+rightChevron.addEventListener('click', rightChevronClick);
+leftChevron.addEventListener('click', leftChevronClick);
+
+var carouselIntervalID = setInterval(autoClick, 3 * 1000);
+
+function rightChevronClick(event) {
+  if (currentIndex === (dataViews.length - 1)) {
+    currentIndex = 0;
+  } else {
+    ++currentIndex;
+  }
+
+  switchViews(dataViews[currentIndex].getAttribute('data-view'));
+  clearInterval(carouselIntervalID);
+}
+
+function autoClick(event) {
+  if (currentIndex === (dataViews.length - 1)) {
+    currentIndex = 0;
+  } else {
+    ++currentIndex;
+  }
+
+  switchViews(dataViews[currentIndex].getAttribute('data-view'));
+}
+
+function leftChevronClick(event) {
+  if (currentIndex === 0) {
+    currentIndex = dataViews.length - 1;
+  } else {
+    --currentIndex;
+  }
+  switchViews(dataViews[currentIndex].getAttribute('data-view'));
+  clearInterval(carouselIntervalID);
```

Comment:
I see you clearing the interval, but I don't see you restarting the interval anywhere. Does your carousel continue after 3 seconds when you click on a chevron or a dot?